camel/storages/object_storages/amazon_s3.py

```diff
@@ -0,0 +1,183 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+import os
+from pathlib import Path, PurePath
+from typing import Optional, Tuple
+
+from colorama import Fore
+
+from camel.loaders import File
+from camel.storages.object_storages.base import BaseObjectStorage
+
+
+class AmazonS3Storage(BaseObjectStorage):
+    r"""A class to connect with AWS S3 object storage to put and get objects
+    from one S3 bucket.
+
+    Args:
+        bucket_name (str): The name of the S3 bucket.
+        create_if_not_exists (bool, optional): Whether to create the bucket if
+            it does not exist. Defaults to True.
+        access_key_id (Optional[str], optional): The AWS access key ID,
+            can be skipped if logged in with AWS CLI. Defaults to None.
+        secret_access_key (Optional[str], optional): The AWS secret access key,
+            can be skipped if logged in with AWS CLI. Defaults to None.
+
+    References:
+        https://aws.amazon.com/pm/serv-s3
+    """
+
+    def __init__(
+        self,
+        bucket_name: str,
+        create_if_not_exists: bool = True,
+        access_key_id: Optional[str] = None,
+        secret_access_key: Optional[str] = None,
+    ) -> None:
+        self._bucket_name = bucket_name
+        self._create_if_not_exists = create_if_not_exists
+
+        aws_key_id = access_key_id or os.getenv("AWS_ACCESS_KEY_ID")
+        aws_secret_key = secret_access_key or os.getenv(
+            "AWS_SECRET_ACCESS_KEY"
+        )
+        if not all([aws_key_id, aws_secret_key]):
+            print(
+                f"{Fore.YELLOW}Warning: AWS access key not configured. "
+                f"Local credentials will be used.{Fore.RESET}"
+            )
+            # make all the empty values None
+            aws_key_id = None
+            aws_secret_key = None
```

Comment:
It's due to the reason that the user can also login through AWS client, who will store the credentials in a local file (and this is also recommended by AWS) and it will be complicated if we also do a check on that. Another case is that the user may want to access a public bucket anonymously

Reply:
Got it. So please add your explanation here as the code commentation, in order to inform others the situation and the inital consideration.